frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/JoinStep.tsx

```diff
@@ -282,7 +282,7 @@ function JoinCondition({
     setOperator,
     setLHSColumn,
     setRHSColumn,
-  } = useJoinCondition(query, stageIndex, table, join, condition);
+  } = useJoinCondition(query, stageIndex, condition);
```

Comment:
There should be useMemo here

Reply:
could you please explain the idea?

Reply:
if I get the idea, I'll create a follow-up PR